purchases/src/main/kotlin/com/revenuecat/purchases/common/offerings/OfferingsFactory.kt

```diff
@@ -87,27 +108,31 @@ internal class OfferingsFactory(
             ProductType.SUBS,
             productIds,
             { subscriptionProducts ->
-                val productsById = subscriptionProducts
-                    .groupBy { subProduct -> subProduct.purchasingData.productId }
-                    .toMutableMap()
-                val subscriptionIds = productsById.keys
+                dispatcher.enqueue(command = {
```

Comment:
I'm wondering if we should do this inside the `BillingWrapper.queryProductDetailsAsync` so it fixes possible problems in other places that call it... But I'm ok doing this for now and rethinking it later.

Reply:
That's true... But that might involve more testing since it might break other parts of the code. I think I am ok with doing this for now, then fixing all calls in BillingWrapper to return in our own threads